### DISCUSSION TITLE
Error in 3D flow field inversion

### DISCUSSION THREAD

**Discussion by 15175829310** [mdolab/dafoam#674] [General]
How should I handle these issues:
1. Why is there only 0 ksp
2. Final error, unable to run
I am looking forward to your reply very much
![error](https://github.com/user-attachments/assets/0bf07d4e-6d2a-4f9f-9824-21d0650ea1e3)
![ksp](https://github.com/user-attachments/assets/48442d1d-ab4a-4e6d-bd21-06900f01a71f)
[logOpt.txt](https://github.com/user-attachments/files/16763023/logOpt.txt)
[runScript_v2.txt](https://github.com/user-attachments/files/16763027/runScript_v2.txt)

**Comment by friedenhe** [comment]
Try `pcFillLevel=1`. Also, set `"adjPartDerivFDStep": {"State": 1e-5}`. 1e-7 is a too small step and can cause problems. So there is no need to change the defaults for adjPartDerivFDStep

**Reply by 15175829310** [reply]
Thank you very much for your help. The previous issue has been resolved, but now there is a new problem.
When I run the PH example, usually every time I calculate 'Evaluating Objective Functions',' Evaluating Objective Functions sensitivities' will appear. However, when I run my example, the program has already calculated 'Evaluating Objective Functions 010', but 'Evaluating Objective Functions 001' has not yet appeared. May I ask what is going on？
[runScript_v2.txt](https://github.com/user-attachments/files/16791508/runScript_v2.txt)
[logOpt.txt](https://github.com/user-attachments/files/16791509/logOpt.txt)

**Comment by friedenhe** [comment]
It is normal to have multiple Evaluating Objective Functions (line searches) before Evaluating Objective Functions Sensitivites. In your case, the flow ran for 1 step and stopped. This is caused by a known bug in our old residual calculation. This issue has been fixed in v3.1.1. Please use the latest version.
